Assets/HoloToolkit/Input/Scripts/InputSources/TouchscreenInputSource.cs

```diff
@@ -0,0 +1,238 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+using UnityEngine;
+using System.Collections.Generic;
+
+namespace HoloToolkit.Unity.InputModule
+{
+    /// <summary>
+    /// Input source supporting basic touchscreen input:
+    /// * taps
+    /// * holds
+    /// Note that a hold-started is raised as soon as a contact exceeds the Epsilon value;
+    /// if the contact subsequently qualifies as a tap then a hold-cancelled is also raised.
+    /// </summary>
+    public class TouchscreenInputSource : BaseInputSource
+    {
+        const float kContactEpsilon = 2.0f/60.0f;
```

Comment:
why not use 1/30?

Reply:
Ha! Coming from several years of VR and now AR dev, my mind tends to get stuck in 60fps mode ;)